ygot/render.go

```diff
@@ -681,6 +698,10 @@ func EncodeTypedValue(val interface{}, enc gnmipb.Encoding) (*gnmipb.TypedValue,
 		if util.IsNilOrInvalidValue(vv) {
 			return nil, nil
 		}
+	default:
+		if underlyingType, ok := unionTypedefUnderlyingTypes[vv.Type().Name()]; ok && vv.Type().ConvertibleTo(underlyingType) {
+			vv = vv.Convert(underlyingType)
+		}
```

Comment:
What happens here if the type is not `ConvertibleTo` the type that we specify in the underlying types -- it looks like we'll fall through to returning the `Interface()`, but if this was an underlying type, isn't that an error case?

(In practice, I assume that this actually isn't possible, because we know that our derived 'string in a union' type is actually just an alias to string?)

Reply:
Good point, I'll add the error return.